rollout/sync.go

```diff
@@ -585,7 +585,7 @@ func (c *rolloutContext) calculateRolloutConditions(newStatus v1alpha1.RolloutSt
 	// In such a case, we should simply not estimate any progress for this rollout.
 	currentCond := conditions.GetRolloutCondition(c.rollout.Status, v1alpha1.RolloutProgressing)
 
-	isCompleteRollout := newStatus.Replicas == newStatus.AvailableReplicas && currentCond != nil && currentCond.Reason == conditions.NewRSAvailableReason
+	isCompleteRollout := newStatus.Replicas == newStatus.AvailableReplicas && currentCond != nil && currentCond.Reason == conditions.NewRSAvailableReason && currentCond.Type != v1alpha1.RolloutProgressing
```

Comment:
I just reviewed the `calculateRolloutConditions` to learn what's going on inside and found one interesting place where I'm a little bit confused:

The next expression `currentCond.Type != v1alpha1.RolloutProgressing` will always be `false` as the `currentCond` is a condition filtered by the v1alpha1.RolloutProgressing type.

That means `isCompleteRollout` is always `false`. Maybe we will remove it?

@jessesuen @huikang Please correct me here if I'm wrong.